clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -483,6 +514,116 @@ public boolean hasUndrained() {
         return false;
     }
 
+
+    private boolean shouldStopDrainBatchesForPartition(ProducerBatch first, TopicPartition tp) {
```

Comment:
I am assuming this method and `drainBatchesForOneNode` is just a refactor? Was any logic changed here? It is hard to tell from the diff and the logic is super intricate so subtle changes are easy to miss. I assume nothing should have to change in this portion of the code since it is totally orthogonal to expiring batches.

Reply:
yes, this is just a refactoring. otherwise, the method `RecordAccumulator.drain` fails in style checking due to too many possible paths in one method.